plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/PodLogsJob.java

```diff
@@ -0,0 +1,157 @@
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.job;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.ui.console.IConsole;
+import org.eclipse.ui.console.IConsoleListener;
+import org.eclipse.ui.console.MessageConsole;
+import org.eclipse.ui.console.MessageConsoleStream;
+import org.jboss.tools.openshift.internal.common.core.job.AbstractDelegatingMonitorJob;
+import org.jboss.tools.openshift.internal.common.ui.OpenShiftCommonUIActivator;
+import org.jboss.tools.openshift.internal.common.ui.console.ConsoleUtils;
+
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.capability.CapabilityVisitor;
+import com.openshift.restclient.capability.resources.IPodLogRetrieval;
+import com.openshift.restclient.model.IPod;
+
+/**
+ * 
+ * @author jeff.cantrill
+ *
+ */
+public class PodLogsJob extends AbstractDelegatingMonitorJob {
+	private static final String DOCUMENT_IS_CLOSED = "Document is closed";
+
+	private static final Map<IPod, ConsoleStreamPipe> REGISTRY = new HashMap<IPod, ConsoleStreamPipe>();
+
+
+	private IPod pod;
+
+	public PodLogsJob(IPod pod) {
+		super("FollowPodLogsJob");
+		this.pod = pod;
+	}
+
+	@Override
+	protected IStatus doRun(IProgressMonitor monitor) {
+		try {
+			monitor.worked(IProgressMonitor.UNKNOWN);
+			if(REGISTRY.containsKey(pod)) {
+				showConsole();
+			}else {
+				ConsoleStreamPipe pipe = pod.accept(new CapabilityVisitor<IPodLogRetrieval, ConsoleStreamPipe>() {
+
+					@Override
+					public ConsoleStreamPipe visit(final IPodLogRetrieval capability) {
+						ConsoleStreamPipe consoleStream = new ConsoleStreamPipe(capability);
+						new Thread(consoleStream).start();
+						return consoleStream;
+					}
+
+				}, null);
+				if(pipe != null) {
+					REGISTRY.put(pod, pipe);
+				}
+			}
+		}finally {
+			monitor.done();
+		}
+		return Status.OK_STATUS;
+	}
+	
+	private MessageConsole showConsole() {
+		final MessageConsole console = ConsoleUtils.findMessageConsole(getMessageConsoleName());
+		ConsoleUtils.displayConsoleView(console);
+		return console;
+	}
+	
+	private String getMessageConsoleName() {
+		return NLS.bind("{0}\\{1} log", pod.getNamespace(), pod.getName());
+	}	
+	
+	private class ConsoleStreamPipe implements Runnable {
+		
+		private IPodLogRetrieval capability;
+		private boolean running = true;
+		
+		ConsoleStreamPipe(IPodLogRetrieval capability){
+			this.capability = capability;
+		}
+		
+		public void stop() {
+			this.running = false;
+			capability.stop();
+		}
+		
+		@Override
+		public void run() {
+			ConsoleUtils.registerConsoleListener(new ConsoleListener(this));
```

Comment:
I think we are at the mercy of what the binary does, and probably is contingent on whether the pod is running.  My brief testing showed that you get a EOF on a finished build and when the build completes.  I presume this is a stopped or stopping pod.  If you have an inactive EAP server, presumably the pod is still running, its just not logging to the stream even though the stream is available.  Also, the listener does some cleanup stuff that we only want when the user clicks the red square which is basically remove the console from view.